railties/lib/rails/generators/rails/app/templates/config/initializers/new_framework_defaults_7_0.rb.tt

```diff
@@ -27,6 +31,10 @@
 # various cache keys leading to cache invalidation.
 # Rails.application.config.active_support.hash_digest_class = OpenSSL::Digest::SHA256
 
+# Don't override ActiveSupport::TimeWithZone.name and use the default Ruby
+# implementation.
+# Rails.application.config.active_support.remove_deprecated_time_with_zone_name = true
```

Comment:
This currently won't work, because this value is checked in a framework initialiser, which runs before application initialisers:

https://github.com/rails/rails/blob/a39098dc856b72a1aff2348b9de805fba71f7b92/activesupport/lib/active_support/railtie.rb#L13

See https://github.com/rails/rails/pull/31547 for a potential solution.

Reply:
I've created an issue: https://github.com/rails/rails/issues/42820